### PR TITLE
Use async header for ManagedApp cmdlet

### PR DESCRIPTION
**Pull request by vivsriaus**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by vivsriaus** [comment]
Tests will follow shortly

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs [hunk 1]
```diff
@@ -101,6 +101,43 @@ internal string WaitOnOperation(OperationResult operationResult)
...
+                    resourceProvisioningState == TerminalProvisioningStates.Ready)
+                    {
+                        return this.GetResourcesClient()
+                            .GetResource<JObject>(resourceId, apiVersion, CancellationToken.None)
+                            .Result
+                            .ToString();
+                    } else if (resourceProvisioningState == TerminalProvisioningStates.Failed ||
+                        resourceProvisioningState == TerminalProvisioningStates.Canceled)
+                    {
+                        var error = JObject.Parse(operationResult.Value);
```
Comment (manoharbp) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: nit: this should be called operationResult

Comment (manoharbp) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: You are doing this on line 105 too. why do it again?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs [hunk 2]
```diff
@@ -101,6 +101,43 @@ internal string WaitOnOperation(OperationResult operationResult)
...
+                        JToken errorToken;
+                        if (error.TryGetValue("error", out errorToken))
+                        {
+                            this.FailedResult(
+                                operationResult,
+                                string.Format("The operation failed because resource is in the: '{0}' state. Please check the logs for more details. The error message: {1}", statusToken, errorToken["message"].ToString()));
```
Comment (manoharbp) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: You should log the whole errorToken and not just errorToken["message"].
Please see the schema for ExtendedErrorInfo in our code

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs [hunk 3]
```diff
@@ -77,12 +77,12 @@ internal LongRunningOperationHelper(string activityName, Func<ResourceManagerRes
         /// Waits for the operation to complete.
         /// </summary>
         /// <param name="operationResult">The operation result.</param>
-        internal string WaitOnOperation(OperationResult operationResult)
+        internal string WaitOnOperation(OperationResult operationResult, string resourceId = null, string apiVersion = null)
```
Comment (manoharbp) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: We should make this change for the Remove appliance cmdlet too

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs (hunk 2 above)
Comment (vivsriaus) [reply src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: This is the response I see. The parent "message" includes all details, including the inner detailed message. What is missing?

{
	"status": "Failed",
	"error": {
		"code": "ApplianceDeploymentFailed",
		"message": "The operation to create appliance failed. Please check operations of deployment 'testmabugapp' under resource group '/subscriptions/38b598fc-e57a-423f-b2e7-dc0ddb631f1f/resourcegroups/mamanagedapp12'. Detailed error messages: '{\r\n  \"error\": {\r\n    \"code\": \"DnsRecordInUse\",\r\n    \"message\": \"DNS record testsdk22212a341s6.westus.cloudapp.azure.com is already used by another public IP.\",\r\n    \"details\": []\r\n  }\r\n};At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.'"
	}
}

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs (hunk 3 above)
Comment (vivsriaus) [reply src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: done

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs (hunk 1 above)
Comment (vivsriaus) [reply src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: This is actually redundant, like you pointed out

Comment (vivsriaus) [reply src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: Fixed

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs (hunk 3 above)
Comment (vivsriaus) [reply src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: Done

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs (hunk 2 above)
Comment (manoharbp) [reply src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs]: For the above, you are probably logging the entire async operation response. Lets sync up f2f for this.

**Comment by maddieclayton** [comment]
@vivsriaus Looks like a few tests are failing: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/4553/artifact/src/Publish/TestResults/AzurePowershellTestResults.html.  Please take a look and assign back to me once they are passing.

**Comment by maddieclayton** [comment]
@vivsriaus Closing for inactivity.  Please reopen when you have looked at the failing tests.

**Pull request closed without merging**